docking.js

```diff
@@ -1652,32 +1687,163 @@ const DockedDash = new Lang.Class({
     },
 
     _showOverlay: function() {
-        if (this._settings.get_boolean('hotkeys-overlay'))
-            this.dash.toggleNumberOverlay(true);
-
-        // Restart the counting if the shortcut is pressed again
-        if (this._numberOverlayTimeoutId) {
-            Mainloop.source_remove(this._numberOverlayTimeoutId);
-            this._numberOverlayTimeoutId = 0;
-        }
+        for (let i = 0; i < allDocks.length; i ++) {
```

Comment:
This is the only part that I really don't like, with each dock being aware and handling other docks. Do we really need this? Can the overlay keyboard shortcut be moved at the dockManager level? Or, can we just have two independent signal listeners both handling the shortcut for their dock?

I couldn't run the code with multiple screens: what happens to the number shortcuts? Does it take into consideration which is the active display, in particular with the isolated workspace option?